src/main/java/com/redhat/devtools/intellij/lsp4ij/client/LSPCompletableFuture.java

```diff
@@ -54,12 +54,15 @@ public ResultOrError(R result, Exception error) {
     private final IndexAwareLanguageClient languageClient;
     private final String progressTitle;
     private final AtomicInteger nbAttempt;
+
+    private final Object coalesceBy;
     private CancellablePromise<ResultOrError<R>> nonBlockingReadActionPromise;
 
-    public LSPCompletableFuture(Function<ProgressIndicator, R> code, String progressTitle, IndexAwareLanguageClient languageClient) {
+    public LSPCompletableFuture(Function<ProgressIndicator, R> code, String progressTitle, IndexAwareLanguageClient languageClient, Object coalesceBy) {

```

Comment:
why Object and not a CoalesceByKey?

Reply:
because perhaps some services don't need that and have an object which defines an uniq hashcode